common/include/pcl/common/impl/centroid.hpp

```diff
@@ -56,27 +56,31 @@ template <typename PointT, typename Scalar> inline unsigned int
 compute3DCentroid (ConstCloudIterator<PointT> &cloud_iterator,
                    Eigen::Matrix<Scalar, 4, 1> &centroid)
 {
-  // Initialize to 0
-  centroid.setZero ();
-
   unsigned cp = 0;
 
+  // Initialize to 0
+  Eigen::Vector4f accu = Eigen::Vector4f::Zero ();
```

Comment:
Using Vector3 would be better. Also, any reason for using float over `Scalar` here?

Reply:
ok fixed it. I guess the only reason would be to avoid casting the `getVector3fMap` to Scalar each time. But it's cast back to Scalar at the end anyway. I've changed it now.

Reply:
I've updated this for the first function. If it looks good, I'll change the rest and eventually squash the commits.

Reply:
The extra cast is an explicit opt-in by the caller of the function. The change looks food to me, please propagate to the rest of the patch

Reply:
done!